### PR TITLE
Mypy coverage

### PR DESCRIPTION
**Pull request by Dreamsorcerer**
Provide some better Mypy coverage. Enables some more options that are useful and adds examples/ into the checker (which helps ensure our public API is correct, and helps keep the examples themselves up-to-date).

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -74,7 +74,7 @@ fmt format:
 
 .PHONY: mypy
 mypy:
-	mypy --show-error-codes aiohttp tests
+	mypy
```
Comment (Dreamsorcerer) [inline Makefile:77]: With no arguments, it uses the files and options in the config file.

File: examples/fake_server.py [hunk 2]
```diff
@@ -50,7 +51,7 @@ def __init__(self) -> None:
                 web.get("/v2.7/me/friends", self.on_my_friends),
             ]
         )
-        self.runner = None
+        self.runner = web.AppRunner(self.app)
```
Comment (Dreamsorcerer) [inline examples/fake_server.py:54]: This example has clearly not been run in several years. :P

File: .mypy.ini [hunk 3]
```diff
@@ -1,24 +1,45 @@
 [mypy]
-warn_unused_configs = True
-strict = True
+files = aiohttp, examples, tests
+check_untyped_defs = True
+follow_imports_for_stubs = True
+#disallow_any_decorated = True
```
Comment (Dreamsorcerer) [inline .mypy.ini:5]: These 2 commented out, we should consider in the future, but will need a ~50 errors to be cleaned up or ignored first.

**Review by webknjaz — APPROVED** [review]

**Comment by webknjaz** [comment]
@Dreamsorcerer would you take care of backporting plz?

**Pull request merged**